llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -433,6 +433,44 @@ using CandidateQueue =
     PriorityQueue<InlineCandidate, std::vector<InlineCandidate>,
                   CandidateComparer>;
 
+// Profile matching statstics.
+class ProfileMatchStats {
```

Comment:
I was mostly thinking about a simple struct to encapsulate different stats, while keeping most of the logic in the Matcher. But don't have strong opinion.